src/shared/varlink.c

```diff
@@ -221,11 +221,11 @@ DEFINE_PRIVATE_STRING_TABLE_LOOKUP_TO_STRING(varlink_state, VarlinkState);
         log_debug("%s: " fmt, varlink_server_description(s), ##__VA_ARGS__)
 
 static inline const char *varlink_description(Varlink *v) {
-        return strna(v ? v->description : NULL);
+        return (v ? v->description : NULL) ?: "varlink";
```

Comment:
why not just `v ? v->description : "varlink"` here?

Reply:
Because v->description may be NULL, and we want to always give pass a string.

Reply:
Ah, was missing that bit of info, thanks

Reply:
@keszybz coverity seems to think `v` can never be null here? False positive?

```
*** CID 1453988:  Control flow issues  (DEADCODE)
/src/shared/varlink.c: 224 in varlink_description()
218             log_debug_errno(error, "%s: " fmt, varlink_server_description(s), ##__VA_ARGS__)
219     
220     #define varlink_server_log(s, fmt, ...) \
221             log_debug("%s: " fmt, varlink_server_description(s), ##__VA_ARGS__)
222     
223     static inline const char *varlink_description(Varlink *v) {
>>>     CID 1453988:  Control flow issues  (DEADCODE)
>>>     Execution cannot reach the expression "NULL" inside this statement: "return (v ? v->description ...".
224             return (v ? v->description : NULL) ?: "varlink";
225     }
226     
227     static inline const char *varlink_server_description(VarlinkServer *s) {
228             return (s ? s->description : NULL) ?: "varlink";
229     }
```

Reply:
In *current* code, this is true. This is a `static` function and all current callers call `varlink_log` with a non-null argument. But my patch didn't change this: the old code was doing the same check. And I think it is correct to do the check: this is how we generally write our logging helpers, e.g. `log_unit_…()`, and the compiler will optimize it away. So I think we should suppress the coverity warning as "intentional".

Reply:
Makes sense, and I agree - marked as intentional and ignored. Thanks!